clients/java/src/main/java/io/camunda/zeebe/client/impl/http/TypedApiEntityConsumer.java

```diff
@@ -91,10 +92,18 @@ public JsonApiEntityConsumer(
 
     @Override
     public ApiEntity<T> generateContent() throws IOException {
-      buffer.asParserOnFirstToken();
-
       if (isResponse) {
-        return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+        try {
+          return ApiEntity.of(json.readValue(buffer.asParserOnFirstToken(), type));
+        } catch (final IOException ioe) {
+          return ApiEntity.of(
+              new ProblemDetail()
+                  .title("Unexpected server response")
+                  .status(500)
+                  .detail(
+                      json.writeValueAsString(
+                          json.readValue(buffer.asParserOnFirstToken(), Map.class))));
```

Comment:
💭 I'm not really happy with this. Overall, I'm wondering if we should go for a similar approach as the raw entity and simply collect the text. That way, we can create a very generic catch clause for the whole `generateContent` method (including trying to parse the problem detail response) that is less prone to run into response obfuscation again. Currently, this can still happen if the response cannot be parsed into a `Map` either.

Reply:
Alternative I will pursue: Flush the original JSON collected in the `TokenBuffer` out into an output stream "raw"